repo/blob/s3/s3_storage.go

```diff
@@ -306,13 +311,28 @@ func newStorage(ctx context.Context, opt *Options) (*s3Storage, error) {
 		return nil, errors.Errorf("bucket %q does not exist", opt.BucketName)
 	}
 
-	return &s3Storage{
+	s := s3Storage{
 		Options:           *opt,
 		cli:               cli,
 		sendMD5:           0,
 		downloadThrottler: downloadThrottler,
 		uploadThrottler:   uploadThrottler,
-	}, nil
+		storageConfig:     &StorageConfig{},
+	}
+
+	if _, err = s.GetMetadata(ctx, ConfigName); err == nil {
```

Comment:
one round-trip to the server (GetMetadata) can be avoided if you do:

err := s.GetBlob(ConfigName)
if err == nil {
  // Load()
} else if !errors.Is(err, blob.ErrBlobNotFound) {
  return errors.Wrap(err, "error retrieving storage config...")
}

Reply:
BTW you also want to add logic to skip `ConfigName` inside `ListBlobs()` - this way the isEmpty check will be fine. Basically the provider will pretend to be empty if only .storageconfig file is in the bucket.

Reply:
Was my original plan, but I didn't know how to check error type. Thanks.